server.go

```diff
@@ -3126,3 +3126,50 @@ func (s *server) watchChannelStatus() {
 		}
 	}
 }
+
+// isNodePublic attempts to determine if a node with the given public key is
+// publicly advertised within the network.
+func (s *server) isNodePublic(pubKey *btcec.PublicKey) (bool, error) {
+	// First, we'll attempt to retrieve the node from the graph.
+	remoteNode, err := s.chanDB.ChannelGraph().FetchLightningNode(pubKey)
+	if err != nil {
+		return false, err
+	}
+
+	// In order to determine whether the node is publicly advertised within
+	// the graph, we'll need to look at all of its edges and check whether
+	// they extend to any other node than us. errDone will be used to
+	// terminate the check early.
+	nodeIsPublic := false
+	ourPub := s.identityPriv.PubKey().SerializeCompressed()
+	errDone := errors.New("done")
+
+	err = remoteNode.ForEachChannel(nil, func(_ *bolt.Tx,
```

Comment:
Seems like this can be simplified down to checking whether we have a node announcement for the node?

Reply:
That's insufficient as we'll still have a node announcement for nodes that we have _private_ channels with.

Reply:
Interesting, looking at the gossiper it seems like we are not checking this before forwarding incoming node announcements, meaning we potentially are leaking their presence there as well.

Reply:
I'm also wondering if it makes sense for a node to send a node announcement if it plans to stay off the radar.

Reply:
Agreed. We should send node announcements also for private channels, but only to the channel counterparty (non-broadcast).

Reply:
Fixed!